addons/calendar/models/res_users.py

```diff
@@ -75,17 +78,17 @@ def write(self, vals):
     @api.depends("res_users_settings_id.calendar_default_privacy")
     def _compute_calendar_default_privacy(self):
         for user in self:
-            user.calendar_default_privacy = user.res_users_settings_id.calendar_default_privacy
+            user.calendar_default_privacy = user.res_users_settings_id.calendar_default_privacy or self._default_user_calendar_default_privacy()
```

Comment:
self._default_user_calendar_default_privacy() value could be stored outside the loop as it does not depends on the user value.

Reply:
Great catch! :clap: :clap: 
Updating it!